UnitTests/GitCommands.Tests/Git/GitCommandHelpersTest.cs

```diff
@@ -205,6 +205,26 @@ public void TestGetDiffChangedFilesFromString()
                 Assert.IsTrue(status[2].Name == " no trim space2 ");
             }
 
+            {
+                // git diff -M -C -z --name-status 123 456
+                // Check renames with spaces
+                string statusString = "R100\0CONTRIBUTING.md\0 CONTRIBUTI NG.md\0C70\0apa.md\0 apa.md\0A\0 co decov.yml\0D\0CODE_OF_CONDUCT.md\0";
+                var status = GitCommandHelpers.GetDiffChangedFilesFromString(module, statusString, "123", "456", null);
+                Assert.IsTrue(status.Count == 4);
+                Assert.IsTrue(status[0].OldName == "CONTRIBUTING.md");
+                Assert.IsTrue(status[0].Name == " CONTRIBUTI NG.md");
+                Assert.IsTrue(status[0].RenameCopyPercentage == "100");
+                Assert.IsTrue(status[0].IsRenamed == true);
+                Assert.IsTrue(status[1].OldName == "apa.md");
+                Assert.IsTrue(status[1].Name == " apa.md");
+                Assert.IsTrue(status[1].RenameCopyPercentage == "70");
+                Assert.IsTrue(status[1].IsCopied == true);
+                Assert.IsTrue(status[2].Name == " co decov.yml");
+                Assert.IsTrue(status[2].IsNew == true);
+                Assert.IsTrue(status[3].Name == "CODE_OF_CONDUCT.md");
+                Assert.IsTrue(status[3].IsDeleted == true);
+            }
```

Comment:
I suggest we use ApprovelTests to assert the blob here.
The code would be something like the following, and that should just generate an \*.approved.\* file that you commit:
```suggestion
                string statusString = "R100\0CONTRIBUTING.md\0 CONTRIBUTI NG.md\0C70\0apa.md\0 apa.md\0A\0 co decov.yml\0D\0CODE_OF_CONDUCT.md\0";
                var status = GitCommandHelpers.GetDiffChangedFilesFromString(module, statusString, "123", "456", null);
                Approvals.Verify(status, "Names");
            }
```

Reply:
I have spent the entire evening on this and may got it working with VerifyAll (Verify is derecated method but there has not been new ApprovalTest updates in 6 years so new versions may not be a problem). Will probably take some days until I am able to finish this.

Reply:
No rush, hola if you get stuck and need a hand

Reply:
Could you please explain the issue?

Reply:
See https://github.com/gerhardol/gitextensions/blob/bugfix/i6723-filename-space-2/UnitTests/GitCommands.Tests/Git/GitCommandsHelperTest.TestGetDiffChangedFilesFromString.ForScenario.Check_that%20spaces%20are%20not%20trimmed%20in%20file%20names.approved.txt

Only the string output, not the contents

Reply:
@RussKie I will initiate the merge, abort that or open a new issue if you have a better solution

Reply:
You've almost nailed it, a little tweak is required.

Reply:
I can't push into your fork, so here's the diff:
```diff
diff --git a/UnitTests/GitCommands.Tests/Git/GitCommandHelpersTest.cs b/UnitTests/GitCommands.Tests/Git/GitCommandHelpersTest.cs
index 6770b2f22..5f678ab0a 100644
--- a/UnitTests/GitCommands.Tests/Git/GitCommandHelpersTest.cs
+++ b/UnitTests/GitCommands.Tests/Git/GitCommandHelpersTest.cs
@@ -1,11 +1,14 @@
 ﻿using System;
 using System.IO;
+using System.Linq;
 using System.Runtime.CompilerServices;
 using ApprovalTests;
 using ApprovalTests.Namers;
 using GitCommands;
 using GitCommands.Git;
+using GitCommands.Utils;
 using GitUIPluginInterfaces;
+using Newtonsoft.Json;
 using NUnit.Framework;
 using ResourceManager;
 
@@ -122,43 +125,31 @@ public void TestFetchArguments()

         public void TestGetDiffChangedFilesFromString(string testName, string firstRevision, string secondRevision, string parentToSecond, string statusString)
         {
             // TODO produce a valid working directory
             var module = new GitModule(Path.GetTempPath());
-            using (ApprovalResults.ForScenario(testName))
+            using (ApprovalResults.ForScenario(testName.Replace(' ', '_')))
             {
                 // git diff -M -C -z --name-status
-                Approvals.VerifyAll<GitItemStatus>(GitCommandHelpers.GetDiffChangedFilesFromString(module, statusString, firstRevision,
-                    secondRevision, parentToSecond), "status");
+                var statuses = GitCommandHelpers.GetDiffChangedFilesFromString(module, statusString, firstRevision, secondRevision, parentToSecond);
+                Approvals.VerifyJson(JsonConvert.SerializeObject(statuses));
             }
         }

```

```diff
diff --git a/UnitTests/GitCommands.Tests/GitCommands.Tests.csproj b/UnitTests/GitCommands.Tests/GitCommands.Tests.csproj
index 459104224..7431b2e38 100644
--- a/UnitTests/GitCommands.Tests/GitCommands.Tests.csproj
+++ b/UnitTests/GitCommands.Tests/GitCommands.Tests.csproj
@@ -1,6 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <ItemGroup>
     <PackageReference Include="JetBrains.Annotations" Version="2018.2.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
   </ItemGroup>
 
   <ItemGroup>
@@ -31,13 +32,18 @@
     </None>
     <EmbeddedResource Include="UserRepositoryHistory\Legacy\MockData\CategorisedRepositories01.xml" />
     <EmbeddedResource Include="UserRepositoryHistory\Legacy\MockData\CategorisedRepositories02.xml" />
-    <None Include="UserRepositoryHistory\RepositoryXmlSerialiserTests.Deserialize_remote_repositories_without_ns.approved.txt" />
-    <None Include="UserRepositoryHistory\RepositoryXmlSerialiserTests.Deserialize_remote_repositories_with_ns.approved.txt" />
-    <None Include="UserRepositoryHistory\RepositoryXmlSerialiserTests.Serialize_recent_repositories.approved.xml" />
-    <None Include="UserRepositoryHistory\Legacy\RepositoryCategorySerialiserTests.Verify_backwards_compatibility_of_object_graph.approved.xml" />
+    <None Include="**\*.approved.*" />
   </ItemGroup>
+  
   <ItemGroup>
-    <None Include="GitRevisionSummaryBuilderTests.Should_do_ellipsis.ForScenario.Too_long_lines.approved.txt" />
-    <None Include="GitRevisionSummaryBuilderTests.Should_do_ellipsis.ForScenario.Too_many_lines.approved.txt" />
+    <None Update="**\*.approved.*">
+      <DependentUpon>$(
+  [System.String]::Copy('%(FileName)')
+    .SubString(
+      0,
+      $([System.String]::Copy('%(FileName)').IndexOf('.'))
+    )
+)$(ProjectExt.Replace('proj', ''))</DependentUpon>
+    </None>
   </ItemGroup>
 </Project>
\ No newline at end of file
```

Reply:
Updated